### PR TITLE
Integration test module

### PR DESCRIPTION
**Pull request by zsoltk**
@vanniktech

### CODE REVIEW

**Comment by zsoltk** [comment]
One of the builds failed for reasons mentioned earlier.

(Also, why are there two distinct travis builds, pr and push, and not one?)

**Comment by vanniktech** [comment]
@zsoltk about the two distinct travis builds, have a [look here](https://github.com/travis-ci/travis-ci/issues/3241#issuecomment-77120970) (I also didn't know about it before)

File: paperwork-integration/build.gradle [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+
+    lintOptions {
+        textReport true
```
Comment (vanniktech) [inline paperwork-integration/build.gradle:54]: We could also add

``` groovy
warningsAsErrors true
abortOnError true
```

File: paperwork-integration/src/androidTest/java/hu/supercluster/paperwork/integration/test/ApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package hu.supercluster.paperwork.integration.test;
+
+import android.app.Application;
+import android.test.ApplicationTestCase;
+
+/**
+ * <a href="http://d.android.com/tools/testing/testing_android.html">Testing Fundamentals</a>
+ */
+public class ApplicationTest extends ApplicationTestCase<Application> {
```
Comment (vanniktech) [inline paperwork-integration/src/androidTest/java/hu/supercluster/paperwork/integration/test/ApplicationTest.java]: can be removed

File: paperwork-integration/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <uses-permission android:name="android.permission.DISABLE_KEYGUARD"/>
+    <uses-permission android:name="android.permission.WAKE_LOCK"/>
+
+    <application
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme"
+        tools:ignore="AllowBackup,GoogleAppIndexingWarning">
```
Comment (vanniktech) [inline paperwork-integration/src/main/AndroidManifest.xml]: we could remove AllowBackup here and just add `fullBackupContent="false"`

**Comment by vanniktech** [comment]
I tried googling the stacktrace message and it does seem like the device when it's booted up still has the lock.

I'd try changing 

``` yaml
- echo no | android create avd --force -n test -t android-18 --abi armeabi-v7a
- emulator -avd test -no-audio -no-window &
- android-wait-for-emulator
- adb shell date -s `date +"%Y%m%d.%H%M%S"`
- adb shell input keyevent 82 &
```

to

``` yaml
- echo no | android create avd --force -n test -t android-18 --abi armeabi-v7a
- emulator -avd test -no-skin -no-audio -no-window &
- android-wait-for-emulator
- adb shell date -s `date +"%Y%m%d.%H%M%S"`
- adb shell input keyevent 82
```

**Comment by zsoltk** [comment]
@vanniktech thank you!

Your idea with the keyevent seems to work! (I hope I wasn't too early on that...)

**Comment by vanniktech** [comment]
You're welcome. I would have squashed all of the commits into one and then merge it. Now the git history is a bit messy. 

On January 14, 2016 11:57:18 PM GMT+01:00, Zsolt Kocsi notifications@github.com wrote:

> @vanniktech thank you!
> 
> Your idea with the keyevent seems to work! (I hope I wasn't too early
> on that...)
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/zsoltk/paperwork/pull/32#issuecomment-171808377

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by zsoltk** [comment]
It's deliberate. Some changes are better understood later if they have their own commit messages.

**Pull request merged**